src/pybind/mgr/cephadm/configchecks.py

```diff
@@ -636,9 +637,10 @@ def _check_kernel_version(self) -> None:
 
     def _process_hosts(self) -> None:
         self.log.debug(f"processing data from {len(self.mgr.cache.facts)} hosts")
-        for hostname in self.mgr.cache.facts:
+        facts = copy.deepcopy(self.mgr.cache.facts)
```

Comment:
According to 

https://github.com/ceph/ceph/blob/2e1c4e590afb566e16f4f5417b7a9037524126b3/src/pybind/mgr/cephadm/inventory.py#L541

```suggestion
        facts = list(self.mgr.cache.facts)
```

Should be enough here.

Reply:
facts is a dict?